snapshots/types/snapshotter.go

```diff
@@ -1,16 +1,26 @@
 package types
 
-import "io"
+import (
+	protoio "github.com/gogo/protobuf/io"
+)
 
 // Snapshotter is something that can create and restore snapshots, consisting of streamed binary
 // chunks - all of which must be read from the channel and closed. If an unsupported format is
 // given, it must return ErrUnknownFormat (possibly wrapped with fmt.Errorf).
 type Snapshotter interface {
 	// Snapshot creates a state snapshot, returning a channel of snapshot chunk readers.
-	Snapshot(height uint64, format uint32) (<-chan io.ReadCloser, error)
+	Snapshot(height uint64, protoWriter protoio.WriteCloser) error
```

Comment:
- format removal: `Snapshotter` choose current best format autonomously, no need to pass in from caller.

Reply:
`Snapshotter` announces the format it uses to snapshot with method `SnapshotFormat()`.